articles/introduction-to-hapi.js-framework/index.md

```diff
@@ -0,0 +1,136 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/introduction-to-hapi/
+title: Introduction to hapi.js Framework
+description: Hapi.js(derived from Http-API) is an open-source Node.js framework used to build powerful and scalable web applications.
+author: peter-kayere
+date: 2020-08-14T00:00:00-07:00
+topics: [Node.js]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/introduction-to-hapi/hero.jpg
+    alt: socket image example
+---
+[Node.js](https://nodejs.com) has become increasingly popular in the software industry. With this popularity comes a great need for developers to create frameworks that make Node.js application development easier. There are various frameworks currently available such as [express](https://expressjs.com), [hapi.js](https://hapi.dev) and [Koa](https://koajs.com) just to mention a few. This article will focus on one of the major Node.js frameworks available i.e. hapi.js.
+<!--more-->
+### What is hapi.js
+Hapi.js(derived from Http-API) is an open-source Node.js framework used to build powerful and scalable web applications. Hapi is commonly used to build Application Programming Interface servers, HTTP-proxy applications, and websites.
+Hapi.js was created by the mobile team at Walmart Labs — led by Eran Hammer to handle their traffic for events like Black Friday, which is by far one of the busiest days for online shopping in the U.S. calendar.
+Hapi was originally built using express framework before facing challenges that drove Walmart to make hapi its own stand-alone framework.
+
+### Hapi.js features
+Hapi.js comes with many unique features that enable developers to build secure, powerful, and scalable applications.
+Some of these features include:
+1. End-to-end code hygiene which helps developers write manageable, controllable, and distributable code.
+2. Secure defaults which are updated regularly. Hapi blocks error messages that may leak information or echo back exploits.
+3. Encrypted and signed cookies, secret or key rotation, and HTTP security headers all meant to enhance the security of applications.
+4. An extensive set of official plugins that are meant to replace middleware used in frameworks such as express.
+5. Integrated Authorization and Authentication Architecture which is the most comprehensive authorization API available in Node.js.
+
+### Who uses hapi.js
+Many companies use hapi.js framework for their websites and web APIs.
+Below is a shortlist of five globally recognized companies that use hapi.js framework.
+- Commercetools
+- Brainhub.
+- Beam.
+- PayPal.
+- Clinlife.
+
+### Creating a Server with hapi.js
+Creating a server using hapi.js is quite easy and obviously different from other frameworks.
+Let's see how we can create a simple server using hapi.
+
+First run
+
+```
+npm init
+```
+To create a node package with the name of your choice.
+
+Then open a new terminal and navigate to your app's folder and run
```

Comment:
These commands are out of order. You'd need to open a terminal and create a directory for your app before running npm init and the npm install.

Reply:
Noted... Let me make the changes right away.